cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -69,9 +70,10 @@ func CreateAwsManager(configReader io.Reader) (*AwsManager, error) {
 
 	service := autoscaling.New(session.New())
 	manager := &AwsManager{
-		asgs:     make([]*asgInformation, 0),
-		service:  service,
-		asgCache: make(map[AwsRef]*Asg),
+		asgs:                     make([]*asgInformation, 0),
+		service:                  service,
+		asgCache:                 make(map[AwsRef]*Asg),
+		instancesNotInManagedAsg: make(map[AwsRef]struct{}),
```

Comment:
@andrewsykim Thanks for the confirmation 👍 

I'm going to include the work of extracting `asgRegistry` as I've described above in this PR.

However, regarding the periodic `regenerateCache` you've suggested, I'm not yet sure why it is needed.
AFAIK, cache regeneration is needed and done only when CA sees a k8s node for the first time. If we agree to assume a k8s node to not move among different ASGs, which seems reasonable to me, we won't need to periodically invalidate the asg cache.
What do you think?